infra/main.bicep

```diff
@@ -165,6 +171,20 @@ module openai './shared/cognitiveservices.bicep' = {
   }
 }
 
+module documentIntelligence './shared/cognitiveservices.bicep' = {
+  name: azureAIDocumentIntelligenceResourceName
+  scope: rg
+  params: {
+    name: azureAIDocumentIntelligenceResourceName
+    location: location
```

Comment:
Things may have changed, is document intelligence available in all regions? Could we get in problems if someone picks a region that it doesn't exist? I believe we do some checking in this repo for Semantic Search as that is not available everywhere

Reply:
Good spot, thanks @ross-p-smith. I have updated this to align with the approach taken for Semantic Search.

https://github.com/microsoft/rag-experiment-accelerator/pull/599/commits/b0d7fb8b655461f260dea30f2bfaf2a518bfd6b9